src/app/guards/guest/guest.guard.ts

```diff
@@ -0,0 +1,25 @@
+import { Injectable } from '@angular/core';
+import { CanActivate, Router, UrlTree } from '@angular/router';
+
+import { Observable } from 'rxjs';
+import { map, tap } from 'rxjs/operators';
+
+import { AppRoutePath } from '@app/enums/app-route-path.enum';
+import { AuthService } from '@app/services/auth/auth.service';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class GuestGuard implements CanActivate {
+  constructor(private router: Router, private authService: AuthService) { }
+
+  canActivate(): Observable<boolean | UrlTree> | Promise<boolean | UrlTree> | boolean | UrlTree {
+    return this.authService.isAuthenticated$.pipe(
+      tap(isAuthenticated => {
+        if (isAuthenticated) {
+          this.router.navigate([ AppRoutePath.Courses ]);
+        }
+      }),
+      map(isAuthenticated => !isAuthenticated));
```

Comment:
check `iif` operator if it can help here.

Reply:
I guess no, it's not very suitable case for using of `iif` operator because:
1. `iif` accepts boolean as first parameter (we have Observable<boolean>)
2. `iif` accepts 2nd and/or 3rd parameters as Observables. We don't have appropriate Observables in our case